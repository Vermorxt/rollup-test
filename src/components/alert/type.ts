import { Ui_colors } from '../_types/colors/ui_colors'
import { HTMLProps, MouseEventHandler } from 'react'

export interface Ui_AlertProps extends HTMLProps<HTMLDivElement>, Ui_colors {
  noIcon?: boolean
  onDenyButtonText?: string
  onAcceptButtonText?: string
  onDeny?: MouseEventHandler<HTMLButtonElement> | undefined
  onAccept?: MouseEventHandler<HTMLButtonElement> | undefined
}
