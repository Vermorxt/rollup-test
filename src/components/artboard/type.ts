import { Ui_colors } from './../_types/colors/ui_colors'
import { HTMLProps } from 'react'

export interface Ui_ArtboardProps extends HTMLProps<HTMLSpanElement>, Ui_colors {
  phone1?: boolean
  phone2?: boolean
  phone3?: boolean
  phone4?: boolean
  phone5?: boolean
  phone6?: boolean
  horizontal?: boolean
  demo?: boolean
}
