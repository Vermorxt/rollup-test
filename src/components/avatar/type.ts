import { HTMLProps } from 'react'
import { Ui_Sizes } from '../_types/sizes/sizes'

interface Ui_Avatar_radius {
  roundedXl?: boolean
  rounded2xl?: boolean
  rounded3xl?: boolean
  roundedLg?: boolean
  roundedFull?: boolean
}

interface Ui_Avatar_ring {
  ring?: boolean
  ringPrimary?: boolean
  ringSecondary?: boolean
  ringOffset?: '0' | '1' | '2' | '4' | '8'
  ringOffsetBackground?: string // NOTE: https://tailwindcss.com/docs/ring-offset-color
}

interface Ui_Avatar_shapes {
  squircle?: boolean
  hexagon?: boolean
  triangle?: boolean
}

interface Ui_Avatar_online {
  online?: boolean
  offline?: boolean
}

export interface Ui_AvatarProps
  extends HTMLProps<HTMLDivElement>,
    Ui_Sizes,
    Ui_Avatar_radius,
    Ui_Avatar_shapes,
    Ui_Avatar_online,
    Ui_Avatar_ring {
  src?: string
  usePlaceHolder?: string // NOTE: content of last avatar item (e.g.: +99)
}
