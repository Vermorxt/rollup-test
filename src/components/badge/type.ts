import { HTMLProps } from 'react'
import { Ui_colors } from '../_types/colors/ui_colors'
import { Ui_Sizes } from '../_types/sizes/sizes'

export type TooltipPositionProps = 'top' | 'left' | 'bottom' | 'right'

export interface Ui_Badge_variants {
  outline?: boolean
  active?: boolean
  disabled?: boolean
  glass?: boolean
}

export interface Ui_Badge_singleClass {
  loading?: boolean
  noAnimation?: boolean
}

export interface Ui_BadgeProps
  extends HTMLProps<HTMLSpanElement>,
    Ui_colors,
    Ui_Sizes,
    Ui_Badge_variants,
    Ui_Badge_singleClass {
  wide?: boolean
  full?: boolean
}
