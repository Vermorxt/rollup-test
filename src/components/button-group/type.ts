import { HTMLProps } from 'react'

export interface Ui_ButtonGroupProps extends HTMLProps<HTMLDivElement> {}
