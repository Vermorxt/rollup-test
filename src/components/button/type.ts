import { HTMLProps } from 'react'
import { Ui_colors } from '../_types/colors/ui_colors'
import { Ui_Sizes } from '../_types/sizes/sizes'
import { TooltipProps } from '../_types/tooltip/tooltip-props'

export interface Ui_Button_variants {
  outline?: boolean
  active?: boolean
  disabled?: boolean
  glass?: boolean
}

export interface Ui_Button_singleClass {
  loading?: boolean
  noAnimation?: boolean
}

export interface Ui_Button_sizes extends Ui_Sizes {
  wide?: boolean
  block?: boolean
  circle?: boolean
  square?: boolean
  full?: boolean
}

export type Ui_ButtonType = 'button' | 'submit' | 'reset' | undefined

export interface Ui_ButtonProps
  extends HTMLProps<HTMLButtonElement>,
    Ui_colors,
    Ui_Button_sizes,
    Ui_Button_variants,
    Ui_Button_singleClass {
  tooltip?: TooltipProps
  type?: Ui_ButtonType
}
