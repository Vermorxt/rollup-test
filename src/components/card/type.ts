import { HTMLProps } from 'react'
import { Ui_colors } from '../_types/colors/ui_colors'
interface Ui_Card_style {
  bgPrimary?: boolean
  bgSecondary?: boolean
  bgNeutral?: boolean
  textPrimaryContent?: boolean
  textSecondaryContent?: boolean
  textNeutralContent?: boolean
  bgBase?: '100' | '200' | '300'
}
export interface Ui_CardProps extends HTMLProps<HTMLDivElement>, Ui_Card_style {
  imageAsBackground?: boolean
  compact?: boolean
  itemsCenter?: boolean
  textCenter?: boolean
  glass?: boolean
  cardSide?: boolean
  cardSideLarge?: boolean
  w96?: boolean
  actionCenter?: boolean
  actionStart?: boolean
  actionSpaceBetween?: boolean
}
