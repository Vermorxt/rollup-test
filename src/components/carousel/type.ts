import { CSSProperties, HTMLProps } from 'react'
export interface Ui_CarouselProps extends HTMLProps<HTMLDivElement> {
  center?: boolean
  vertical?: boolean
  end?: boolean
  full?: boolean
  fullHeight?: boolean
  halfWidth?: boolean
  rounded?: boolean
  relative?: boolean
}
export interface Ui_CarouselArrowNavigationProps {
  index: number
  itemLength: number
  usedId: string
  style?: CSSProperties
  className?: string
}
