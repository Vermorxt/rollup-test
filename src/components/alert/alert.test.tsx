import React from 'react'
import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import { checkAccessibility, itRendersChildren, itSupportsSystemProps } from '../../_tests-utils'
import { itHasButtonAndCallbackFired } from '../../_tests-utils/it-has-button-and-callback-fired'
import { itHasCssClass } from '../../_tests-utils/it-has-color-css-class'
import { Ui_AlertProps } from './type'
import Ui_Alert from './alert'

const defaultProps: Ui_AlertProps = {
  className: 'test',
  noIcon: false,
  onDenyButtonText: 'deny',
  onAcceptButtonText: 'ok',
  onDeny: () => false,
  onAccept: () => true,
}

const selector = 'alert'

describe('@cms/core/Ui_Alert', () => {
  itRendersChildren(Ui_Alert, defaultProps as any)
  checkAccessibility([<Ui_Alert {...defaultProps} ref={null} />])
  itSupportsSystemProps({
    component: Ui_Alert,
    props: defaultProps,
    refType: HTMLDivElement,
  })

  // NOTE: (possible debug): debugComponent(<Ui_Alert {...defaultProps} ref={null} />)

  it('should match snapshot', () => {
    const { container } = render(<Ui_Alert {...defaultProps} noIcon ref={null} />)
    expect(container).toMatchSnapshot()
  })

  it('renders icon based on noIcon prop', () => {
    const { container: available } = render(<Ui_Alert {...defaultProps} noIcon ref={null} />)
    const { container: notAvailable } = render(<Ui_Alert {...defaultProps} ref={null} />)

    expect(available.querySelector('.used-alert-icon')).toBeNull()
    expect(notAvailable.querySelectorAll('.used-alert-icon')).toHaveLength(1)
  })
  it('renders ok button', async () => {
    const { container } = render(<Ui_Alert {...defaultProps} ref={null} />)

    const button = screen.getByRole('button', { name: 'ok' })

    expect(button).toBeInTheDocument()
    expect(button).not.toBeDisabled()
  })

  it('renders deny button', async () => {
    const { container } = render(<Ui_Alert {...defaultProps} ref={null} />)

    const button = screen.getByRole('button', { name: 'deny' })

    expect(button).toBeInTheDocument()
    expect(button).not.toBeDisabled()
  })

  it('clicks OK button and has called function', async () => {
    const mockCallback = jest.fn()
    const { container } = render(<Ui_Alert {...defaultProps} ref={null} onAccept={mockCallback} />)

    await itHasButtonAndCallbackFired('ok', mockCallback)
  })

  it('clicks DENY button and has called function', async () => {
    const mockCallback = jest.fn()
    const { container } = render(<Ui_Alert {...defaultProps} ref={null} onDeny={mockCallback} />)

    await itHasButtonAndCallbackFired('deny', mockCallback)
  })

  it('renders bg colors', async () => {
    await itHasCssClass(<Ui_Alert primary />, `.${selector}-primary`)
    await itHasCssClass(<Ui_Alert secondary />, `.${selector}-secondary`)
    await itHasCssClass(<Ui_Alert accent />, `.${selector}-accent`)
    await itHasCssClass(<Ui_Alert info />, `.${selector}-info`)
    await itHasCssClass(<Ui_Alert error />, `.${selector}-error`)
    await itHasCssClass(<Ui_Alert success />, `.${selector}-success`)
    await itHasCssClass(<Ui_Alert warning />, `.${selector}-warning`)
    await itHasCssClass(<Ui_Alert ghost />, `.${selector}-ghost`)
    await itHasCssClass(<Ui_Alert neutral />, `.${selector}-neutral`)
    await itHasCssClass(<Ui_Alert link />, `.${selector}-link`)
  })
})
