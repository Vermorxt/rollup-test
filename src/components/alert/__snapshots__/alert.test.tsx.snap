// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@cms/core/Ui_Alert should match snapshot 1`] = `
<div>
  <div
    class="
      alert shadow-lg
        
        
        
        
        
        
        
        
        
        
      test
      "
  >
    <div />
    <div
      class="flex-none"
    >
      <button
        class="
        btn 
        
        
        
        
        
        
        
        btn-ghost
        
        
        
        
        
        
        
        
        
        btn-sm
        
        
        
        
        
        "
        type="button"
      >
        deny
      </button>
      <button
        class="
        btn 
        btn-primary
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        btn-sm
        
        
        
        
        
        "
        type="button"
      >
        ok
      </button>
    </div>
  </div>
</div>
`;
